src/Metadata/Resource/Factory/FiltersResourceMetadataCollectionFactory.php

```diff
@@ -51,6 +51,11 @@ public function create(string $resourceClass): ResourceMetadataCollection
 
         foreach ($resourceMetadataCollection as $i => $resource) {
             foreach ($operations = $resource->getOperations() ?? [] as $operationName => $operation) {
+                if (null !== $operation->getFilters()) {
+                    // filters are already applied on this operation
+                    continue;
```

Comment:
Aren't you losing the possibility to add filters at the resource level to an operation that has predefined filters here?
If I understant the current version correctly, it's made to let you have the filters option on the operation and resources filters active at the same ttime

Reply:
I think if the operation already has filters, we should not override it, even from resource filters. For example:
```php
#[ApiResource(
    filters: ['foo', 'bar'],
    uriTemplate: '/foos',
    operations: [
        new GetCollection(), // inherit filters and uriTemplate from resource
        new GetCollection(filters: ['bar', 'baz'], uriTemplate: '/baz'), // override filters and uriTemplate
        new GetCollection(filters: [], uriTemplate: '/no-filters'), // disable filters and override uriTemplate
    ]
)]
```

Reply:
It may be the right way to do this, but since it currently merges both your change would be a BC break wouldn't it? 
I'm just saying this should at least be documented if it's changed this way.

Reply:
Indeed, it will change the behavior and it's a BC break. WDYT @soyuka?